lib/order.rb

```diff
@@ -0,0 +1,52 @@
+require_relative 'texting'
+
+class Order
+
+  include Texting
+
+  attr_reader :basket
+
+  def initialize
+    @basket = {}
+  end
+
+  def add_to_basket(menu, dish_number, quantity)
+    dish = menu.select(dish_number)
+    quantity.times do
+      if basket.keys.include?(dish)
+        @basket[dish] += dish.price
+      else
+        @basket[dish] = dish.price
+      end
+    end
+  end
+
+  def summary
+    fail "Basket empty" if empty_basket?
+    basket.each_pair do |k, v|
+      puts "#{(v / k.price).to_i}x #{k.name} | £#{'%.2f' % v}"
+    end
+    puts "Total cost: £#{'%.2f' % total_cost}"
```

Comment:
Style/FormatString: Favor `format` over `String#%`.<br>Rails/Output: Do not write to stdout. Use Rails' logger if you want to log.